### PR TITLE
Storage availability fixes for files_sharing and FailedStorage

### LINKED ISSUE #18455: Bogus shared::$path storage on master
### Steps

1. Create two users "user1" and "user2"
1. Login as "user1"
1. Create a folder "test"
1. Share "test" with "user2"
1. Login as "user2"
1. Check "oc_storages"

### Expected result
No entries for local shares.

### Actual result
An extra entry "shared::test" is created.
```
MariaDB [owncloud]> select * from oc_storages;
+---------------------------------------+------------+-----------+--------------+
| id                                    | numeric_id | available | last_checked |
+---------------------------------------+------------+-----------+--------------+
| home::root                            |          1 |         1 |         NULL |
| local::/srv/www/htdocs/owncloud/data/ |          2 |         1 |         NULL |
| home::user1                           |          3 |         1 |         NULL |
| home::user2                           |          4 |         1 |         NULL |
| shared::/test                         |          5 |         1 |         NULL |
+---------------------------------------+------------+-----------+--------------+
```

### Versions
ownCloud 8.2 pre-alpha git master (d93bf3548226ab509f075a879ff71768e6b454b9)

Regression from 8.1.1

I'm going to bisect this.

### PR DESCRIPTION
**Pull request by RobinMcCorkell**
The first commit prevents local shares from creating oc_storages entries due to updating storage availability (which is irrelevant for shares anyway). The second commit prevents the same happening for FailedStorage from files_external, as that is by definition a failed storage so we don't need to store it in the DB.

Fixes #18455 

Please review and test @PVince81 @MorrisJobke @DeepDiver1975 @icewind1991

### CODE REVIEW

File: apps/files_sharing/lib/sharedstorage.php [hunk 1]
```diff
@@ -662,4 +662,22 @@ public function changeLock($path, $type, ILockingProvider $provider) {
...
+			'available' => true,
+			'last_checked' => 0
+		];
+	}
+
+	/**
+	 * @param bool $available
+	 */
+	public function setAvailability($available) {
+		// shares do not participate in availability logic
```
Comment (PVince81) [inline apps/files_sharing/lib/sharedstorage.php]: they felt that they were too special and opted-out

Comment (PVince81) [inline apps/files_sharing/lib/sharedstorage.php]: too hipster to participate in the availability logic :stuck_out_tongue:

**Comment by PVince81** [comment]
:+1: works

**Comment by MorrisJobke** [comment]
:+1:

File: apps/files_sharing/lib/sharedstorage.php (hunk 1 above)
Comment (Xenopathic) [inline apps/files_sharing/lib/sharedstorage.php]: hah

**Pull request merged**